### PR TITLE
fix: shorten instance id

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: <link>

### Overview

- Reduce the instance id suffix by half.

### Rationale

- GitHub rejects too long instance names.

> The runner name 'stg-runner-openstack-arm64-jammy-edge-ps6-1-56ddd0f47df7bb60cb0fd40d' is not valid. A valid runner name is 64 characters or less in length and does not include '"', '/', ':', '<', '>', '\', '|', '*' and '?'. (Parameter 'agentName')

### Module Changes

- `openstack_runner_manager.py` - shorten instance ID.

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] The library version in `pyproject.toml` is incremented

### CODE REVIEW

**Review by jdkandersson — DISMISSED** [review]

File: src/github_runner_manager/openstack_cloud/openstack_runner_manager.py [hunk 1]
```diff
@@ -736,12 +736,20 @@ def _wait_runner_running(self, instance: OpenstackInstance) -> None:
...
-        """Generate a instance id.
+        r"""Generate an instance id suffix compliant to the GitHub runner naming convention.
+
+        The GitHub runner name convention is as following:
+        A valid runner name is 64 characters or less in length and does not include '"', '/', ':',
+        '<', '>', '\', '|', '*' and '?'.
+
+        The collision rate calculation:
+        alphanumeric 12 chars long (26 alphabet + 10 digits = 36)
+        36^12 is big enough for our use-case.
```
Comment (cbartz) [inline src/github_runner_manager/openstack_cloud/openstack_runner_manager.py:691]: You are changing it to 6 chars, so its `36^6` , isn't it? Could you please update the docstring?

Comment (yanksyoon) [reply src/github_runner_manager/openstack_cloud/openstack_runner_manager.py:691]: It's hex - which = 12 chars

**Review by cbartz — DISMISSED** [review]

**Comment by yanksyoon** [comment]
Not merging this yet, there's a problem w/ the test run.

**Review by amandahla — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]

**Comment by yanksyoon** [comment]
Passing tests; https://github.com/canonical/github-runner-operator/actions/runs/11453672712/job/31928023372

**Pull request merged**